catppuccin_tmux.conf

```diff
@@ -0,0 +1,279 @@
+source -F "#{d:current_file}/themes/catppuccin_#{@catppuccin_flavor}_tmux.conf"
+
+source -F "#{d:current_file}/status/application.conf"
+source -F "#{d:current_file}/status/battery.conf"
+source -F "#{d:current_file}/status/clima.conf"
+source -F "#{d:current_file}/status/cpu.conf"
+source -F "#{d:current_file}/status/date_time.conf"
+source -F "#{d:current_file}/status/directory.conf"
+source -F "#{d:current_file}/status/host.conf"
+source -F "#{d:current_file}/status/load.conf"
+source -F "#{d:current_file}/status/pomodoro_plus.conf"
+source -F "#{d:current_file}/status/session.conf"
```

Comment:
Maybe better to not load all modules by default to reduce load time.

My suggestion would be to only load modules that require _no_ extra plugin since most people will only use those.
For the other modules something like `set -gF @catppuccin_dir "#{d:current_file}"`
and in the use config (`tmux.conf`) `source -F "#{@catppuccin_dir}/status/<module>.conf"`

This would also limit the effect of adding more modules and allow easy sourcing of `status_module.conf` for custom modules.

Reply:
I always prefer to measure before optimising. "premature optimisation is the root of all evil". 

What's the difference in speed? Does it matter? Have you measured it?

Also for custom modules I was thinking of just letting people write their own formats, since it's so easy now.

Reply:
- no modules: 
- basic modules: application, date_time, directory, host, session, uptime, user
- all modules/current: added gitmux, uptime, user and weather on top of what is currently sourced (were missing

![image](https://github.com/user-attachments/assets/dff6811f-1336-43fa-9869-1159f04b08ad)

The cost per module seems to be +1.3 ms (for me) almost negligible but it's more about putting a limit on it so it does not grow out of hand.
![image](https://github.com/user-attachments/assets/6ee21900-501a-4456-ab0c-4c7158cbe6b3)

Reply:
I get putting a limit on it, that's reasonable. I think we discussed that 200ms was a reasonable upper limit? WDYT?

Reply:
We landed on 250ms we are _well_ under that (and don't think we will ever reach it).
I just think if we draw a line on what modules to auto import plugin-less modules is the only one you can really draw.

Thinking about it how would plugins work with tpm?
They need (in most cases) `#{@catppuccin_status_<module>}` to be expanded in `status-left/right` before running but you can only expand after running the catppuccin plugin and tpm runs all plugins in sequence on `run '~/.tmux/plugins/tpm/tpm'`. Am I missing something?

Only solution I can see is in `catppuccin_tmux.conf` detect if `@plugin` is set and expand the necessary options. (or just always expand them).